server/services/addressService.ts

```diff
@@ -5,12 +5,65 @@ import LicenceApiClient from '../data/licenceApiClient'
 export default class AddressService {
   constructor(private readonly licenceApiClient: LicenceApiClient) {}
 
-  async searchForAddresses(searchQuery: string, user: User): Promise<AddressSearchResponse[]> {
-    if (!searchQuery || searchQuery.trim().length === 0) {
-      return []
+  async searchForAddresses(
+    searchQuery: string,
+    user: User,
+    priorityStrings: string[] = [],
+  ): Promise<AddressSearchResponse[]> {
+    if (!searchQuery?.trim()) return []
+
+    const requestBody = { searchQuery }
+    const results = await this.licenceApiClient.searchForAddresses(requestBody, user)
+    if (results.length > 1 && priorityStrings.length > 0) {
+      const addressSortingInfo = this.generateAddressSortingInfo(results, priorityStrings)
+
+      addressSortingInfo.sort(
+        (a, b) =>
+          b.matchCount - a.matchCount ||
+          this.compareAddressPrioritySequences(a.prioritySequence, b.prioritySequence) ||
+          a.originalIndex - b.originalIndex,
+      )
+
+      return addressSortingInfo.map(item => item.address)
+    }
+    return results
+  }
+
+  private generateAddressSortingInfo(results: AddressSearchResponse[], priorityStrings: string[]) {
+    const lowercasePriorities = priorityStrings.map(p => p.toLowerCase())
+
+    return results.map((address, index) => this.rankAddressForSorting(address, index, lowercasePriorities))
+  }
+
+  private rankAddressForSorting(address: AddressSearchResponse, index: number, priorities: string[]) {
+    const addressText = [address.firstLine, address.secondLine ?? ''].join(',').toLowerCase()
+
```

Comment:
have done first line and second as probabtion can turn up on the second line and I also cant find anything call Probation St, Probation Rd